### ISSUE TITLE
VPR crashes with unconnected model outputs

### ISSUE THREAD

**Issue by jackwadden** [verilog-to-routing/vtr-verilog-to-routing#174]
#### Expected Behaviour
VPR should allow output ports in .subckt models to be unconnected (either not defined or routed to unconn), just like inputs.

#### Current Behaviour
If I don't specify an output in the .subckt, I get a nice error message:
** Message: Found no output pin on .subckt 'blah' **

If I specify any output as 'unconn' I get the following failed assertion:
** VTR/vtr_dev/vpr/SRC/base/atom_netlist.c:789 AtomBlockId AtomNetlist::create_block(std::string, const t_model*, AtomNetlist::TruthTable): Assertion 'block_model(blk_id) == model' failed.
Aborted (core dumped) **

If I omit the .names unconn definition I get the following failed assertion:
** VTR/vtr_dev/vpr/SRC/base/atom_netlist.c:876 AtomPinId AtomNetlist::create_pin(AtomPortId, BitIndex, AtomNetId, AtomPinType, bool): Assertion 'pin_net(pin_id) == net_id' failed.
Aborted (core dumped) **

#### Steps to Reproduce (for bugs)
Files to reproduce bug: [http://bit.ly/2hWpECU](http://bit.ly/2hWpECU)

Using VPR: Version: 8.0.0-dev Revision: 880d066
** $ vpr --max_router_iterations 500 --timing_tradeoff 0 --beta_clustering 1 --cluster_seed_type max_inputs --alpha_clustering 0 --timing_analysis off test.xml 1.blif **
Will produce the error message

2.blif has model outputs connected to unconn...
** $ vpr --max_router_iterations 500 --timing_tradeoff 0 --beta_clustering 1 --cluster_seed_type max_inputs --alpha_clustering 0 --timing_analysis off test.xml 2.blif **
...will cause the (block_model(blk_id) == model) assertion to fail.

3.blif has model outputs connected to unconn without a .names unconn definition...
** $ vpr --max_router_iterations 500 --timing_tradeoff 0 --beta_clustering 1 --cluster_seed_type max_inputs --alpha_clustering 0 --timing_analysis off test.xml 3.blif **
...will cause the (pin_net(pin_id) == net_id) assertion to fail.

#### Context
I am trying to successfully place and route circuits that have nodes with unconnected output ports.

#### Your Environment

* VTR revision used: 880d066
* Operating System and version: Ubuntu 14.04.5 LTS
* Compiler version: g++-5 (Ubuntu 5.4.1-2ubuntu1~14.04) 5.4.1 

Thanks!

**Comment by kmurray** [comment]
The crashes should now be fixed in 9d6816f.

It turns out my initial suggestion about using the `unconn` net was not quite correct (it only applied to input pins, not output pins).

For unconnected output pins you will need to leave them implicitly disconnected, or attach them to uniquely named dummy nets with no sinks (which are automatically removed by VPR).

I've documented the details of handling unconnected pins [here](https://docs.verilogtorouting.org/en/latest/vpr/file_formats/#unconnected-primitive-pins).

With the unused ports left implicitly disconnected (see https://gist.github.com/5dc80a1dd2414e0744230d01898f5858) I was able to place and route the circuit with:
> vpr test.xml 3.implicitly_disconnected.blif --timing_analysis off --sweep_dangling_blocks off

Where `--sweep_dangling_blocks off` is required to ensure that the whole netlist doesn't get swept away.
